solutions/usaco-916.mdx

```diff
@@ -0,0 +1,55 @@
+---
+id: usaco-916
+source: USACO Bronze 2019 February
+title: The Great Revegetation
+author: Ryan Chou
+---
+## Solution 1
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_revegetate_bronze_feb19.html)
+
+## Solution 2
+We can iterate through the favorite pastures, and increment one to the pasture with the greater index, making the result minimal.
+We'll also cycle through the previous favorite pastures to make sure that our new alteration doesn't interfere with any of the other cow's favorite pastures.
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N^2\log{}N)$
```

Comment:
isn't the time complexity N^2 or am i missing something?

Reply:
we're sorting the array

Reply:
not at every iteration though, right? so O(N^2 + log N) = O(N^2)?

Reply:
ohhhh, my bad, i read the code incorrectly. sorry about that. 👍